internal/nodes/avi_model_l7_translator.go

```diff
@@ -477,10 +473,23 @@ func (o *AviObjectGraph) BuildPolicyPGPoolsForSNI(vsNode []*AviVsNode, tlsNode *
 			var httpPolicySet []AviHostPathPortPoolPG
 
 			httpPGPath := AviHostPathPortPoolPG{Host: host}
+
+			if path.PathType == networkingv1beta1.PathTypeExact {
+				httpPGPath.MatchCriteria = "EQUALS"
+			} else {
+				// PathTypePrefix and PathTypeImplementationSpecific
+				// default behaviour for AKO set be Prefix match on the path
+				httpPGPath.MatchCriteria = "BEGINS_WITH"
+			}
+
 			if path.Path != "" {
-				httpPGPath.Path = append(httpPGPath.Path, path.Path)
+				httpPolicyPath := path.Path
+				if !strings.HasSuffix(httpPolicyPath, "/") {
+					httpPolicyPath = path.Path + "/"
+				}
+				httpPGPath.Path = append(httpPGPath.Path, httpPolicyPath)
 			}
-			httpPGPath.MatchCriteria = "BEGINS_WITH"
```

Comment:
If PathType is not set then what would be the default MatchCriteria ?

Reply:
pathType None defaults to ImplementationSpecific,  which for us means Prefix matching